src/Core/EnvironmentSettings.cs

```diff
@@ -50,6 +52,15 @@ public int PCHGenerationTimeoutInMs(int defaultValue)
             return userValue > 0 ? userValue : defaultValue;
         }
 
+        public bool LogDebugMessages()
+        {
+            if (bool.TryParse(Environment.GetEnvironmentVariable(LogDebugMessagesEnvVar), out var result))
+            {
+                return result;
+            }
+            return false;
+        }
+
```

Comment:
Same as the `TreatBlockerSeverityAsError` method above.

Reply:
Minor: we have a `ParseInt` method, it might be worth adding `ParseBoolean` just for consistency and to make it a bit shorter. It's already pretty clear though so it's not that important.